packages/server/utils/directories.ts

```diff
@@ -23,7 +23,10 @@ export function directoryWalker({
 }) {
     let cleanedUrl;
     try {
-        cleanedUrl = decodeURI(pathname);
+        cleanedUrl = pathname
```

Comment:
you can just simly use `decodeURIComponent`, no need of doing this amount of garbage, and also test your code with different approaches first

![image](https://github.com/KotRikD/tosu/assets/21735205/85e81758-121f-417b-9e23-9ebbf7b16d67)
![image](https://github.com/KotRikD/tosu/assets/21735205/54e823ce-b591-4c28-9723-a637ad38ab24)

Reply:
`decodeURIComponent` encodes the slash character. I can't shove the entire pathname in there. The code I splits the full URL by slashes "blahblahblah#/blahblah" and encodes each chunk with `decodeURIComponent`, and this works just fine. Not sure what the issue is here.

Reply:
Hmmm, `decodeURI` seems shouldn't be used here, because `decodeURI/encodeURI` is designed to be used in full path URI's. Let's talk about solution with `decodeURIComponent`, @thenerdie can you provide cases, where standard `decodeURIComponent` fails and you needed to use map function?

Reply:
Fixed. I was braindead coding last night LOL